R/package.R

```diff
@@ -178,13 +178,100 @@ pkgdown_config_path <- function(path) {
   )
 }
 
-read_meta <- function(path) {
-  path <- pkgdown_config_path(path)
+read_meta <- function(path, check_path = TRUE) {
+  if (check_path) {
+    # check_path = FALSE can be used to supply a direct path to
+    # read_meta instead of a pkgdown object.
+    path <- pkgdown_config_path(path)
+  }
 
+  # If parsing fails, it will say Error in read_meta()
+  call <- current_env()
   if (is.null(path)) {
     yaml <- list()
   } else {
-    yaml <- yaml::yaml.load_file(path) %||% list()
+    yaml <- withCallingHandlers(
+      yaml::yaml.load_file(path),
+      error = function(e) {
+        yaml_error_msg <- conditionMessage(e)
+        # Create a clickable path, in case we end up
+        # just rethrowing the original message
+        yaml_error_msg <- gsub(
+          "\\(([^\\)]+)\\)",
+          "({.path \\1})",
+          yaml_error_msg
+        )
+
+        # trying to extract the parsing error type?
+        # like block mapping, block collection?
+        what <- regmatches(
+          yaml_error_msg,
+          m = regexpr("block\\s[[:alpha:]]+", yaml_error_msg)
+        )
+        # search for all occurences of digits (1 to 4 digits long)
+        # I would not see a _pkgdown.yml file containing >9999 lines.
+        error_locations <- regmatches(
+          yaml_error_msg,
+          m = gregexpr("\\d{1,4}", yaml_error_msg)
+        )
+        error_locations <- unlist(error_locations, use.names = FALSE)
+        if (length(error_locations) != 4 || !is_string(what)) {
+          # can't parse the error message properly,
+          # just rethrow it cli style
+          cli::cli_abort(yaml_error_msg, call = call)
+        }
```

Comment:
```suggestion
        yaml_error_msg <- conditionMessage(e)
        # Create a clickable path (replaces (path) with ({.path path})
        yaml_error_msg <- gsub(
          "\\(([^\\)]+)\\)",
          "({.path \\1})",
          yaml_error_msg
        )
         # just rethrow it cli style
          cli::cli_abort(yaml_error_msg, call = call)
   ```

The fix could also just be just this: just adding a clickable path and throwing the error cli style.
![Capture d’écran, le 2024-05-18 à 3 57 51 p m](https://github.com/r-lib/pkgdown/assets/52606734/307812f4-860e-4344-9adf-eaf92cc3082c)

Reply:
Opened https://github.com/r-lib/pkgdown/pull/2559 that basically implements this idea. SImpler is better I guess?